Code/GraphMol/MarvinParse/MarvinParser.cpp

```diff
@@ -0,0 +1,1110 @@
+                }
+                break;
+
+              default:
+                throw FileParseException(
+                    "Plus sign coordinate must have 4 MPoints in MRV file");
+            }
+            ++pointCount;
+          }
+        }
+      }
+
+      BOOST_FOREACH (boost::property_tree::ptree::value_type &v, documentTree) {
+        if (v.first != "MTextBox") {
+          continue;
+        }
+
+        auto *marvinCondition = new MarvinCondition();
+        res->conditions.push_back(
+            std::move(std::unique_ptr<MarvinCondition>(marvinCondition)));
+        marvinCondition->id = v.second.get<std::string>("<xmlattr>.id", "");
+        marvinCondition->halign =
+            v.second.get<std::string>("<xmlattr>.halign", "");
+        marvinCondition->valign =
+            v.second.get<std::string>("<xmlattr>.valign", "");
+        double fontScale;
+        std::string fontScaleStr =
+            v.second.get<std::string>("<xmlattr>.fontScale", "");
+        if (fontScaleStr != "") {
+          if (!getCleanDouble(fontScaleStr, fontScale)) {
+            throw FileParseException(
+                "Condition font scale must be a positive integer in MRV file");
+          }
+        } else {
+          fontScale = 0.0;
+        }
+        marvinCondition->fontScale = fontScale;
+
+        marvinCondition->text = v.second.get<std::string>("Field", "");
+
+        int pointCount = 0;
+        BOOST_FOREACH (boost::property_tree::ptree::value_type &v2, v.second) {
+          if (v2.first == "MPoint") {
+            double x, y;
+            std::string xStr = v2.second.get<std::string>("<xmlattr>.x", "");
+            std::string yStr = v2.second.get<std::string>("<xmlattr>.y", "");
+            if (!getCleanDouble(xStr, x) || !getCleanDouble(yStr, y)) {
+              throw FileParseException(
+                  "Condition coordinate must valid integers in MRV file");
+            }
+
+            switch (pointCount) {
+              case 0:  //  first point - x1 and y1 are set
+                marvinCondition->x = x;
+                marvinCondition->y = y;
+                break;
+              case 1:
+              case 2:
+              case 3:  // x and Y are checked - must be the same as point 1
+                break;
+
+              default:
+                throw FileParseException(
+                    "Condition defs must have 4 MPoints in MRV file");
+            }
+            ++pointCount;
+          }
+        }
+      }
+
+      return res;
+    } catch (const std::exception &e) {
+      delete res;
+
+      throw;
+    }
+  }
+};
+
+//------------------------------------------------
+//
+//  Read a molecule from a stream
+//
+//------------------------------------------------
+
+bool MrvDataStreamIsReaction(std::istream &inStream) {
+  PRECONDITION(inStream, "no stream");
+
+  using boost::property_tree::ptree;
```

Comment:
better to have the `using` at the global scope

Reply:
Turns out this isnt used at all, so I delete it - also at a few other places